### PR TITLE
RavenDB-12253: Unused connection string can be deleted now and wont a…

### PR DESCRIPTION
**Pull request by garayx**
…ppear on "Used in Tasks" list.

### CODE REVIEW

**Comment by ppekrol** [comment]
@ml054 please review this

File: src/Raven.Studio/typescript/viewmodels/database/settings/connectionStrings.ts [hunk 1]
```diff
@@ -137,8 +137,10 @@ class connectionStrings extends viewModelBase {
         }    
     }   
 
-    isConnectionStringInUse(connectionStringName: string) :boolean {
-        return _.includes(Object.keys(this.connectionStringsTasksInfo), connectionStringName);       
+    isConnectionStringInUse(connectionStringName: string, task: string): boolean {
+        if (this.connectionStringsTasksInfo[connectionStringName].find(x => x.TaskType === task))
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/settings/connectionStrings.ts]:
```suggestion
        return this.connectionStringsTasksInfo[connectionStringName].find(x => x.TaskType === task));
```

File: src/Raven.Studio/typescript/viewmodels/database/settings/connectionStrings.ts [hunk 2]
```diff
@@ -216,11 +218,16 @@ class connectionStrings extends viewModelBase {
             });
     }
     
-    private getTasksThatUseThisString(connectionStringName: string, taskType: Raven.Client.Documents.Operations.OngoingTasks.OngoingTaskType) : { taskName: string; taskId: number }[]{        
-        const tasksData = this.connectionStringsTasksInfo[connectionStringName];               
-               
-        return tasksData ? _.sortBy(tasksData.map((task) => { return { taskName: task.TaskName, taskId: task.TaskId }; }), 
-                                    x => x.taskName.toUpperCase()) : [];    
+    private getTasksThatUseThisString(connectionStringName: string, taskType: Raven.Client.Documents.Operations.OngoingTasks.OngoingTaskType): { taskName: string; taskId: number }[] {
+        if (this.connectionStringsTasksInfo[connectionStringName] == null)
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/settings/connectionStrings.ts]: in typescript we put { around if every time

File: src/Raven.Studio/typescript/viewmodels/database/settings/connectionStrings.ts [hunk 3]
```diff
@@ -216,11 +218,16 @@ class connectionStrings extends viewModelBase {
...
+            return [];
+        else
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/settings/connectionStrings.ts]: in TS use TS like formating for {, ex.:
```
if (condition) {
  ...
} else {
   ...
}
```

File: src/Raven.Studio/typescript/viewmodels/database/settings/connectionStrings.ts (hunk 2 above)
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/settings/connectionStrings.ts]: if (!this.connectionStringsTasksInfo[connectionStringName])

**Review by ml054 — CHANGES_REQUESTED** [review]

File: src/Raven.Studio/typescript/viewmodels/database/settings/connectionStrings.ts (hunk 2 above)
Comment (garayx) [reply src/Raven.Studio/typescript/viewmodels/database/settings/connectionStrings.ts]: done

File: src/Raven.Studio/typescript/viewmodels/database/settings/connectionStrings.ts (hunk 3 above)
Comment (garayx) [reply src/Raven.Studio/typescript/viewmodels/database/settings/connectionStrings.ts]: done

File: src/Raven.Studio/typescript/viewmodels/database/settings/connectionStrings.ts (hunk 2 above)
Comment (garayx) [reply src/Raven.Studio/typescript/viewmodels/database/settings/connectionStrings.ts]: done

File: src/Raven.Studio/typescript/viewmodels/database/settings/connectionStrings.ts (hunk 1 above)
Comment (garayx) [reply src/Raven.Studio/typescript/viewmodels/database/settings/connectionStrings.ts]: the logic I used here is not right, I have changed this so pls check in next PR.

File: src/Raven.Studio/typescript/viewmodels/database/settings/connectionStrings.ts [hunk 4]
```diff
@@ -137,8 +137,9 @@ class connectionStrings extends viewModelBase {
         }    
     }   
 
-    isConnectionStringInUse(connectionStringName: string) :boolean {
-        return _.includes(Object.keys(this.connectionStringsTasksInfo), connectionStringName);       
+    isConnectionStringInUse(connectionStringName: string, task: string): boolean {
+        return _.includes(Object.keys(this.connectionStringsTasksInfo), connectionStringName)
+            && (typeof this.connectionStringsTasksInfo[connectionStringName].find(x => x.TaskType === task) === 'object');
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/settings/connectionStrings.ts]: I think typeof is not needed here. 

```suggestion
            && this.connectionStringsTasksInfo[connectionStringName].find(x => x.TaskType === task);
```

File: src/Raven.Studio/typescript/viewmodels/database/settings/connectionStrings.ts [hunk 5]
```diff
@@ -216,11 +217,16 @@ class connectionStrings extends viewModelBase {
...
     
-    private getTasksThatUseThisString(connectionStringName: string, taskType: Raven.Client.Documents.Operations.OngoingTasks.OngoingTaskType) : { taskName: string; taskId: number }[]{        
-        const tasksData = this.connectionStringsTasksInfo[connectionStringName];               
-               
-        return tasksData ? _.sortBy(tasksData.map((task) => { return { taskName: task.TaskName, taskId: task.TaskId }; }), 
-                                    x => x.taskName.toUpperCase()) : [];    
+    private getTasksThatUseThisString(connectionStringName: string, taskType: Raven.Client.Documents.Operations.OngoingTasks.OngoingTaskType): { taskName: string; taskId: number }[] {
+        if (!this.connectionStringsTasksInfo[connectionStringName]) {
+            return [];
+        }
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/settings/connectionStrings.ts]: it is still not valid javascript formatting:

if (..) {
   ..
} else {    // this goes in single line
    ...
}

Comment (garayx) [reply src/Raven.Studio/typescript/viewmodels/database/settings/connectionStrings.ts]: done

File: src/Raven.Studio/typescript/viewmodels/database/settings/connectionStrings.ts (hunk 4 above)
Comment (garayx) [reply src/Raven.Studio/typescript/viewmodels/database/settings/connectionStrings.ts]: done

**Review by ml054 — APPROVED** [review]
LGTM

**Pull request merged**